### ISSUE TITLE
Support regex for selector-no-qualifying-type

### ISSUE THREAD

**Issue by alexilyaev** [stylelint/stylelint#1908]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Missing regex option for `selector-no-qualifying-type`

> Which rule, if any, is this issue related to?

`selector-no-qualifying-type`

> What CSS is needed to reproduce this issue?

```css
li {
  &.selected {
    color: red;
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "selector-no-qualifying-type": [true, {
      "ignore": ["^&\\..+"]
    }]
  }
}
```

> Which version of stylelint are you using?

`7.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint --syntax scss 'app/**/*.scss'`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

no

> What did you expect to happen?

Would like to disallow qualifying but allow nested rules in SCSS using `&`

> What actually happened (e.g. what warnings or errors you are getting)?

No regex option

**Comment by jeddy3** [comment]
@alexilyaev Thanks for using the issue template.

I don't think I understand your request though. There is no *qualifying type selector* in your CSS snippet:

```css
.foo {
  &.bar {}
}
```

It is resolved to `.foo.bar`, which is two *class* selectors. As such, it currently doesn't warn and the rules behaves as you describe.

Can you expand on your original request please.

**Comment by alexilyaev** [comment]
@jeddy3 You are right, it actually does what it says and I didn't understand it correctly :-).
Closing.

Btw, I'm really glad so many options have been added, makes my life much easier now.
Thanks!
